library/core/src/cell/once.rs

```diff
@@ -10,6 +10,12 @@ use crate::{fmt, mem};
 ///
 /// For a thread-safe version of this struct, see [`std::sync::OnceLock`].
 ///
+/// A OnceCell conceptually has two states, called the `uninitialized state`
+/// and the `initialized state`.
+///
+/// Like an `enum { Uninitialized, Initialized(T) }`,
+/// except that it has invariants to uphold, so the enum is hidden.
```

Comment:
If anything, I would leave this as something like:

```suggestion
/// A `OnceCell` can be thought of as a safe abstraction over uninitialized data that becomes
/// initialized once written.
```

Since uninitialized vs. initialized is reasonably fundamental in programming, probably easier to understand than Rust enums.

If you keep it, move it above the "For a thread-safe version..." line.